src/clients/c++/experimental_api_v2/library/cencode.c

```diff
@@ -102,7 +102,7 @@ int base64_encode_blockend(char* code_out, base64_encodestate* state_in)
 	case step_A:
 		break;
 	}
-	*codechar++ = '\n';
+	*codechar++ = '\0';
```

Comment:
It would be better not to change this file. It seems that we just need to be careful to always use the returned size and not rely on an end-of-string indicator. It seems you made those changes below so can we no do this change.